Datum/tests/Integration/demo3.tests.ps1

```diff
@@ -0,0 +1,40 @@
+if ($PSScriptRoot) {
+    $here = $PSScriptRoot
+}
+else {
+    $here = Join-Path $pwd.Path '*\tests\Integration\' -Resolve
+}
+
+Write-verbose "Here: $here"
+
+Describe 'Test Datum overrides' {
+    Context 'Most specific Merge behavior' {
+        BeforeAll {
+
+            $Datum = New-Datumstructure -DefinitionFile  (Join-path $here '.\assets\Demo3\Datum.yml' -Resolve)
+
+            $AllNodes = @($Datum.AllNodes.psobject.Properties | ForEach-Object {
+                $Node = $Datum.AllNodes.($_.Name)

```

Comment:
I think your problem is here (and, to be honest, because of the lack of documentation!).
Your `$Node` is missing the property `Name`, which is the BaseName of the file.
See this: https://github.com/gaelcolas/Datum/blob/master/Datum/tests/Integration/experiments.ps1#L13-L15

Using some of your data without the code it works for me without any change:
```PowerShell
$Node = @{} + $Datum.AllNodes.DEV.SRV01
$Node['Environment'] = 'DEV'
$Node['Name'] = 'SRV01'

Lookup -PropertyPath 'Disks' -Node $Node -DatumTree $Datum -options @{'Disks' = @{merge_hash_array = 'UniqueKeyValTuples'; merge_options = @{tuple_keys= @('Number')} }} -Verbose
```
That returns me the 3 disks.
It's getting late, so I'll sleep on it and will push some data tomorrow to illustrate your use case. But I don't see a problem with Datum (so far).